src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -511,7 +511,7 @@ private void ScanSyntaxToken(ref TokenInfo info)
 
                 case '!':
                     TextWindow.AdvanceChar();
-                    if (TextWindow.PeekChar() == '=')
+                    if (TextWindow.PeekChar() == '=' && TextWindow.PeekChar(-2) != '!')
```

Comment:
I don't believe this is correct. `myFunctionWithANullableWarning!!=1` would be a valid expression.

Reply:
As that could be either a valid expression, or a parameter and we can't determine that from the lexer, we may have to take the same approach we do for `>` here: parse them as separate, and glue the `!=` together if we're in that context.

---
In reply to: [466642902](https://github.com/dotnet/roslyn/pull/46520#discussion_r466642902) [](ancestors = 466642902)

Reply:
Yup, removed in latest commit.